File: kjplayer/src/main/java/com/mic/kjplayer/BaseActivity.java
package com.mic.kjplayer;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
    }
}

File: khmplayer/src/main/java/com/mic/khmplayer/base/BaseActivity.kt
package com.mic.khmplayer.base

import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity

abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (isFullScreen()) {
            //全屏设置
            //全屏设置
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
                window.addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)
            }
        }
        setContentView(getLayoutId())
        initListener()
        initData()
    }

    open abstract fun getLayoutId(): Int

    open protected fun initListener() {}

    open protected fun initData() {}


    open abstract fun isFullScreen(): Boolean

    inline fun <reified T : BaseActivity> startActivity() {
        val intent = Intent(this, T::class.java)
        this.startActivity(intent)
        this.finish()
    }

}